internal/pkg/fleetapi/action.go

```diff
@@ -555,6 +676,16 @@ func (a *Actions) UnmarshalYAML(unmarshal func(interface{}) error) error {
 					"fail to decode CANCEL_ACTION action",
 					errors.TypeConfig)
 			}
+		case ActionTypeDiagnostics:
```

Comment:
@michel-laterman I think a similar handler is missing from the `UnmarshalJSON` function, when I try request diagnostics from the UI, I see this in the logs:
```
[elastic_agent][error] HandlerUnknown: action 'action_id: 272cdd37-9ac8-4061-bc7f-dd74602f98b6, type: UNKNOWN (original type: REQUEST_DIAGNOSTICS)' received
```

Reply:
good catch

Reply:
@michel-laterman thanks for fixing. Does this work for you locally? You can test from the UI if you enable the feature flag [here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/fleet/common/experimental_features.ts#L18).

<img width="1616" alt="image" src="https://user-images.githubusercontent.com/90178898/214521896-81a9cdf0-2a00-4cc0-9a23-e2aaf410a440.png">